file.c

```diff
@@ -8286,19 +8284,28 @@ doExternal(URLFile uf, char *type, Buffer *defaultbuf)
 {
     Str tmpf, command;
     struct mailcap *mcap;
-    int mc_stat;
+    int mc_stat, err;
     Buffer *buf = NULL;
     char *header, *src = NULL, *ext = uf.ext;
+    FILE *ff;
 
     if (!(mcap = searchExtViewer(type)))
 	return NULL;
 
-    if (mcap->nametemplate) {
-	tmpf = unquote_mailcap(mcap->nametemplate, NULL, "", NULL, NULL);
-	if (tmpf->ptr[0] == '.')
-	    ext = tmpf->ptr;
+    if ((mcap->flags & MAILCAP_NOSAVEINPUT)) {
+	if (uf.scheme == SCM_LOCAL) {
+	    tmpf = Strnew_charp(uf.url);
+	} else {
+	    tmpf = Strnew_charp("-"); /* make it a pipe */
```

Comment:
IIUC, this uses '-' as file name for the command. This assumes every command understands '-' to mean "Read from stdin".  It this standardized? Could we do it w/o this assumption and let the user add it if it's needed to make the command work?